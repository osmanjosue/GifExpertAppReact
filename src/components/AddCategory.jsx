import { useState } from 'react';
import {PropTypes} from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("One Punch");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    /* console.log('Hola Mundo desde OnSubmit') */
    event.preventDefault(); //evita el refresh para que se muestre el clg de la sig linea

    if (inputValue.trim().length <= 1) return;

    // setCategories((categories) => [inputValue, ...categories]);
    setInputValue('');
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
