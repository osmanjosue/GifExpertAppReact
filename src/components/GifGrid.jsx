import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const [Images, setImages] = useState([]);

  const getImages = async() => {
    const newImages = await getGifs(category);
    setImages(newImages);
  }

  useEffect(() => {
    getImages();
        /* .then(newImages => setImages(newImages)) */
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <ol>
        {
            Images.map(({title, id})=> {
                return <li key={id}>{title}</li>
            })
        }
      </ol>
    </>
  );
};
