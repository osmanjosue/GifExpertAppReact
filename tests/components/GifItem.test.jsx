import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("pruebas en GifItem", () => {
  const title = "Saitama";
  const url = "https://one-punch.com/saitama.jpg";

  test("should make match with the snapshot", () => {
    const {container}=render(<GifItem title={title} url={url} />);
    expect ( container ).toMatchSnapshot();
  });
});

