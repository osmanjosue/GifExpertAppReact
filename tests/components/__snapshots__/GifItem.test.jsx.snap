// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem should make match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
